build-recipe-spec

```diff
@@ -126,91 +187,97 @@ recipe_build_spec() {
+		return
+	    fi
+	    : > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
+	    if test "$st" = 11 ; then
+		reqsfile=
+		for i in "$BUILD_ROOT$TOPDIR/SRPMS/"*.buildreqs.nosrc.rpm ; do
+		    test -f "$i" && reqsfile=${i##*/}
+		done
+		test -n "$reqsfile" || cleanup_and_exit 1 "no buildreqs.nosrc.rpm file?"
+		chroot $BUILD_ROOT rpm -qp --requires "$TOPDIR/SRPMS/$reqsfile" | grep -v '^rpmlib(' | sort -u > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp
+		chroot $BUILD_ROOT rpmspec -q --srpm --requires "$TOPDIR/SOURCES/$RECIPEFILE" >> $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp || cleanup_and_exit 1 "rpm -q --spec failed"
+		sort < $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp | uniq -u > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
+		rm -f $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp
+		while read db ; do
+		    GEN_BUILDREQS_PACKS[${#GEN_BUILDREQS_PACKS[@]}]="$db"
+		done < $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
+	    fi
+	    rm -f "$BUILD_ROOT$TOPDIR/SRPMS/"*.buildreqs.nosrc.rpm
+	fi
+
+	# su involves a shell which would require even more
+	# complicated quoting to bypass than this
 	toshellscript $rpmbuild \
-		--nodeps \
 		"${definesnstuff[@]}" \
-		"${rpmdynbropts[@]}" \
+		"${rpmbopts[@]}" \
 		"$TOPDIR/SOURCES/$RECIPEFILE" \
 		> $BUILD_ROOT/.build.command
 	chmod 755 $BUILD_ROOT/.build.command
+	check_exit
 	chroot $BUILD_ROOT su -c /.build.command - $BUILD_USER < /dev/null
 	st=$?
-	if test "$st" != 0 -a "$st" != 11 ; then
-	    return
-	fi
-	: > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
-	if test "$st" = 11 ; then
-	    reqsfile=
-	    for i in "$BUILD_ROOT$TOPDIR/SRPMS/"*.buildreqs.nosrc.rpm ; do
-		test -f "$i" && reqsfile=${i##*/}
-	    done
-	    test -n "$reqsfile" || cleanup_and_exit 1 "no buildreqs.nosrc.rpm file?"
-	    chroot $BUILD_ROOT rpm -qp --requires "$TOPDIR/SRPMS/$reqsfile" | grep -v '^rpmlib(' | sort -u > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp
-	    chroot $BUILD_ROOT rpmspec -q --srpm --requires "$TOPDIR/SOURCES/$RECIPEFILE" >> $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp || cleanup_and_exit 1 "rpm -q --spec failed"
-	    sort < $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp | uniq -u > $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
-	    rm -f $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs_tmp
-	    while read db ; do
-		GEN_BUILDREQS_PACKS[${#GEN_BUILDREQS_PACKS[@]}]="$db"
-	    done < $BUILD_ROOT$TOPDIR/OTHER/_generated_buildreqs
-	fi
-	rm -f "$BUILD_ROOT$TOPDIR/SRPMS/"*.buildreqs.nosrc.rpm
-    fi
-
-    # su involves a shell which would require even more
-    # complicated quoting to bypass than this
-    toshellscript $rpmbuild \
-	    "${definesnstuff[@]}" \
-	    "${rpmbopts[@]}" \
-	    "$TOPDIR/SOURCES/$RECIPEFILE" \
-	    > $BUILD_ROOT/.build.command
-    chmod 755 $BUILD_ROOT/.build.command
-    check_exit
-    chroot $BUILD_ROOT su -c /.build.command - $BUILD_USER < /dev/null
-    st=$?
+	test "$st" = 0 -o "$st" = 11 || break
```

Comment:
Why do you also test for `11` here?

Reply:
Right, on 11 the caller should probably handle this. I will fix this.